src/FileSystem/FileSystem.cpp

```diff
@@ -559,13 +560,11 @@ bool CFileSystem::extract(const std::string& filename,
 			delete archive;
 			return false;
 		}
-		int res = 1;
-		if (!buf.empty())
-			res = fwrite(&buf[0], buf.size(), 1, f);
+		size_t written = fwrite(buf.data(), 1, buf.size(), f); // might create empty file
```

Comment:
instead of 1 should be 
`sizeof(decltype(buf)::value_type);`
but I don't think we have c++11?